contracts/governancestaking/BaseGovernanceShareField.sol

```diff
@@ -57,8 +57,9 @@ contract BaseGovernanceShareField is DAOContract {
         uint256 multiplier = block.number - lastRewardBlock; // Blocks passed since last reward block
         uint256 reward = multiplier * rewardsPerBlock; // rewardsPerBlock is in GDAO which is in 0 decimals
 
-        accAmountPerShare =accAmountPerShare + rdiv(reward * 1e2 ,totalProductivity) / 1e25; //multiply reward by 1e2 so bring it to 2 decimals as well then divide result by 1e25 so accAmountPerShare is in 2 decimals
-        lastRewardBlock = block.number;
+        accAmountPerShare = accAmountPerShare + rdiv(reward ,totalProductivity * 1e16) / 1e9; // totalProductivity in 2decimals since it is GD so we multiply it by 1e16 to bring 18 decimals then rdiv and  divide result of it to 1e9 so reduce it to 18decimals
```

Comment:
accAmountPerShare should probably be kept in 1e27 because reward<totalProductivity would lead to 0

Reply:
verify in regular staking also

Reply:
Changed for BaseGovernanceShareField also need to check for regular one